src/test/java/org/jabref/logic/crawler/CrawlerTest.java

```diff
@@ -0,0 +1,47 @@
+package org.jabref.logic.crawler;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.jabref.logic.exporter.SavePreferences;
+import org.jabref.logic.importer.ParseException;
+import org.jabref.logic.util.io.FileUtil;
+import org.jabref.model.entry.BibEntryTypesManager;
+import org.jabref.model.metadata.SaveOrderConfig;
+import org.jabref.model.util.DummyFileUpdateMonitor;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Answers;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class CrawlerTest {
+    @TempDir
+    Path tempRepositoryDirectory;
+
+    @Test
+    public void performCrawl() throws IOException, ParseException {
+        SavePreferences preferences = mock(SavePreferences.class, Answers.RETURNS_DEEP_STUBS);
+        when(preferences.getSaveOrder()).thenReturn(new SaveOrderConfig());
+        when(preferences.getEncoding()).thenReturn(null);
+        when(preferences.takeMetadataSaveOrderInAccount()).thenReturn(true);
+        BibEntryTypesManager entryTypesManager = new BibEntryTypesManager();
+        Crawler testCrawler = new Crawler(Paths.get("C:\\Users\\Dominik\\Documents\\Studium\\Informatik Semester 8 Uni Stuttgart\\Bachelorarbeit\\TestStudyRepository"),
```

Comment:
I think, this introduces a dependency to an external repository. I can creawte https://github.com/jabref/jabref-slr-test-repository - and we add a git submodule here.

Alternatively, the test reposiory is created at the start of the cralwer test `@Before`. 1. create git repository in temp dir, 2. add files from `src/test/resources`, ...